### ISSUE TITLE
aarch64 / GB10: cuCheckpointProcessCheckpoint returns CUDA_ERROR_NOT_SUPPORTED (is arm64 support planned?)

### ISSUE THREAD

**Issue by rocker-zhang** [NVIDIA/cuda-checkpoint#51]
The prebuilt `cuda-checkpoint` binary in this repo is x86_64 only (`bin/x86_64_Linux/`), and the driver-level checkpoint API appears only partially implemented on aarch64. This reports what works and what doesn't on a Grace Blackwell box and asks whether arm64 support is planned.

Environment:
- NVIDIA GB10 (DGX Spark), aarch64
- Driver 580.142 (NVIDIA UNIX Open Kernel Module for aarch64)
- CUDA 13.0

Calling the driver checkpoint APIs directly (no CLI, no CRIU) with a process checkpointing itself, the way [src/r580-migration-api.c](src/r580-migration-api.c) does:

```
cuInit                          -> 0   (no error)
cuDevicePrimaryCtxRetain        -> 0   (no error)
cuCtxSetCurrent                 -> 0   (no error)
cuMemAlloc / cuMemcpyHtoD       -> 0   (no error)
cuCheckpointProcessGetState     -> 0   (RUNNING)
cuCheckpointProcessLock         -> 0   (state -> LOCKED)
cuCheckpointProcessCheckpoint   -> 801 (CUDA_ERROR_NOT_SUPPORTED)
cuCheckpointProcessUnlock       -> 0   (no error)
```

Lock, Unlock, and GetState all succeed, and ordinary CUDA work runs fine; only the checkpoint step returns 801. The result is the same whether or not GPU memory was allocated beforehand. 801 (CUDA_ERROR_NOT_SUPPORTED) is a capability error rather than a permission one, so this doesn't look like the root/persistence-mode condition that affects `--toggle`.

Minimal repro:

```c
#include <stdio.h>
#include <unistd.h>
#include <cuda.h>
#define P(call) do { CUresult r=(call); const char*s=0; cuGetErrorString(r,&s); \
  printf("%-32s -> %d (%s)\n", #call, (int)r, s?s:"?"); } while(0)
int main(void){
  setvbuf(stdout,0,_IONBF,0);
  CUcontext ctx; pid_t self=getpid();
  P(cuInit(0));
  P(cuDevicePrimaryCtxRetain(&ctx,0));
  P(cuCtxSetCurrent(ctx));
  CUcheckpointLockArgs la={0}; CUcheckpointCheckpointArgs ca={0}; CUcheckpointUnlockArgs ua={0};
  P(cuCheckpointProcessLock(self,&la));
  P(cuCheckpointProcessCheckpoint(self,&ca));
  P(cuCheckpointProcessUnlock(self,&ua));
  return 0;
}
```

```
gcc -I /usr/local/cuda/include cp_diag.c -o cp_diag -lcuda && ./cp_diag
```

Questions:
- Is checkpoint/restore expected to work on aarch64 / Grace Blackwell, or is it x86_64 only at the driver level for now?
- If arm64 is planned, is there a target driver release? If it's intentionally unsupported right now, mentioning that in the README would help, since the README currently says nothing about architecture.

**Comment by rocker-zhang** [comment]
Correcting my own report: the README already documents this. Under "Functionality" it notes that `cuda-checkpoint` "is x64 only" and that these limitations "will be addressed in subsequent display driver releases," since the utility exposes functionality contained in the driver. I missed that section when filing this, so the architecture question and the README suggestion are already covered.

Recording one detail for the record before closing: on aarch64 (GB10, driver 580.142) the API surface is partially present rather than absent. `cuCheckpointProcessLock`, `cuCheckpointProcessUnlock`, and `cuCheckpointProcessGetState` all return CUDA_SUCCESS and ordinary CUDA work runs; only `cuCheckpointProcessCheckpoint` returns 801 (CUDA_ERROR_NOT_SUPPORTED). Closing as already documented.
